src/tools/remote-test-server/src/main.rs

```diff
@@ -356,6 +368,17 @@ fn my_copy(src: &mut dyn Read, which: u8, dst: &Mutex<dyn Write>) {
     }
 }
 
+fn batch_copy(buf: &[u8], which: u8, dst: &Mutex<dyn Write>) {
+    let n = buf.len();
+    let mut dst = dst.lock().unwrap();
+    t!(dst.write_all(&[which, (n >> 24) as u8, (n >> 16) as u8, (n >> 8) as u8, (n >> 0) as u8,]));
```

Comment:
As a drive-by improvement since we're changing this code, can we replace the manual shifting with writing `(n as u32).encode_le_bytes()` and `read_u32` to use `u32::from_le_bytes`?

Reply:
(But let's do that in a separate PR if you're up to it).